plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -121,13 +128,33 @@ export class HogTransformerService {
                 const teamHogFunctions = await this.hogFunctionManager.getHogFunctionsForTeam(event.team_id, [
                     'transformation',
                 ])
+
                 const transformationResult = await this.transformEvent(event, teamHogFunctions)
                 await this.hogFunctionMonitoringService.processInvocationResults(transformationResult.invocationResults)
 
+                const messagePromises: Promise<void>[] = [this.hogFunctionMonitoringService.produceQueuedMessages()]
+
+                const shouldRunHogWatcher = Math.random() < this.hub.CDP_HOG_WATCHER_SAMPLE_RATE
+
+                if (shouldRunHogWatcher) {
+                    const timer = hogWatcherLatency.startTimer({ operation: 'observeResults' })
+                    messagePromises.push(
+                        this.hogWatcher
+                            .observeResults(transformationResult.invocationResults)
+                            .then(() => {
+                                timer() // Stop the timer when the promise resolves
+                            })
+                            .catch((error) => {
+                                timer() // Make sure to stop the timer even if there's an error
+                                logger.warn('⚠️', 'HogWatcher observeResults failed', { error })
+                            })
+                    )
+                }
+
                 hogTransformationCompleted.inc({ type: 'with_messages' })
                 return {
                     ...transformationResult,
-                    messagePromises: [this.hogFunctionMonitoringService.produceQueuedMessages()],
+                    messagePromises,
```

Comment:
Not a biggy but this should be renamed to soemthing like `scheduledPromises` or whatever. It's not "message" specific and I feel like that was what led you down the wrong path the first time.